src/crimson/osd/watch.cc

```diff
@@ -51,4 +51,66 @@ seastar::future<> Watch::start_notify(NotifyRef notify)
   return is_connected() ? send_notify_msg(*it) : seastar::now();
 }
 
+seastar::future<> Watch::notify_ack(
+  const uint64_t notify_id,
+  const ceph::bufferlist& reply_bl)
+{
+  logger().info("{}", __func__);
+  return seastar::do_for_each(in_progress_notifies,
+    [this_shared=shared_from_this(), &reply_bl] (auto notify) {
+      return notify->complete_watcher(this_shared, reply_bl);
+    }
+  ).then([this] {
+    in_progress_notifies.clear();
+    return seastar::now();
+  });
+}
+
+seastar::future<> Notify::complete_watcher(
+  WatchRef watch,
+  const ceph::bufferlist& reply_bl)
+{
+  if (discarded || complete) {
+    return seastar::now();
+  }
+#if 0
+  notify_replies.emplace_back(watch->get_watcher_gid(),
+                              watch->get_cookie(),
+                              reply_bl);
```

Comment:
could use:

```c++
  notify_replies.emplace(make_pair(watch->get_watcher_gid(),
                                   watch->get_cookie()),
                         reply_bl);
```

and drop the `#if #else #endif`

Reply:
I would prefer to replace this code with dedicated `struct` to avoid `std::pair`.